shared/actions/teams.js

```diff
@@ -593,6 +593,48 @@ const _getTeamPublicity = function*(action: TeamsGen.GetTeamPublicityPayload): S
   yield Saga.put(createDecrementWaiting({key: Constants.teamWaitingKey(teamname)}))
 }
 
+function _getChannelInfo(action: TeamsGen.GetChannelInfoPayload) {
+  const {teamname, conversationIDKey} = action.payload
+  const waitingKey = {key: Constants.getChannelInfoWaitingKey(teamname, conversationIDKey)}
+  return Saga.all([
+    Saga.call(RPCChatTypes.localGetInboxAndUnboxUILocalRpcPromise, {
+      identifyBehavior: RPCTypes.tlfKeysTLFIdentifyBehavior.chatGui,
+      query: ChatConstants.makeInboxQuery([conversationIDKey]),
+    }),
+    Saga.identity(teamname),
+    Saga.identity(conversationIDKey),
+    Saga.identity(waitingKey),
+    Saga.put(createIncrementWaiting(waitingKey)),
+  ])
```

Comment:
you can likely use sequentially and do waiting as the first and last things here to avoid all the branching in the _afterGetChannelInfo